gradle-jdks-setup-common/src/main/java/com/palantir/gradle/jdks/setup/common/GradleJdksPatchHelper.java

```diff
@@ -69,30 +68,23 @@ public static List<String> getLinesWithoutPatch(List<String> initialLines) {
         return linesNoPatch;
     }
 
-    public static void maybeRemovePatch(Path filePath) {
+    public static void writeContentWithPatch(
+            Path outputPath, List<String> initialLines, List<String> patchLines, int insertIndex) {
         try {
-            List<String> initialLines = Files.readAllLines(filePath);
-            List<String> linesWithoutPatch = getLinesWithoutPatch(initialLines);
-            if (!linesWithoutPatch.equals(initialLines)) {
-                Files.write(
-                        filePath,
-                        String.join("\n", linesWithoutPatch).getBytes(StandardCharsets.UTF_8),
-                        StandardOpenOption.TRUNCATE_EXISTING,
-                        StandardOpenOption.WRITE);
-            }
+            Files.write(
+                    outputPath,
+                    getContentWithPatch(initialLines, patchLines, insertIndex).getBytes(StandardCharsets.UTF_8));
         } catch (IOException e) {
-            throw new RuntimeException(String.format("Unable to read file %s", filePath), e);
+            throw new RuntimeException("Unable to write file", e);
         }
     }
 
-    public static byte[] getContentWithPatch(List<String> initialLines, List<String> patchLines, int insertIndex) {
+    private static String getContentWithPatch(List<String> initialLines, List<String> patchLines, int insertIndex) {
         List<String> newLines = new ArrayList<>(initialLines.size() + patchLines.size());
         newLines.addAll(initialLines.subList(0, insertIndex));
         newLines.addAll(patchLines);
         newLines.addAll(initialLines.subList(insertIndex, initialLines.size()));
-        return newLines.stream()
-                .collect(Collectors.joining(System.lineSeparator()))
-                .getBytes(StandardCharsets.UTF_8);
+        return newLines.stream().collect(Collectors.joining(System.lineSeparator())) + "\n";
```

Comment:
I think we need to check whether the existing file had a newline or not? Unless we're just depending on `./gradlew` always having a newline.

Reply:
addressed in [e268e40](https://github.com/palantir/gradle-jdks/pull/465/commits/e268e40a0a07ed44914204e08e561467ba2f1974), but I think in general it is nice for files to end with a newline and I think it is safe to assume that `./gradlew` will always have a new line